### PR TITLE
feat(admin): rename Dashboard to Overview, add summary cards (#338)

### LINKED ISSUE #338: Admin Overview: rename dashboard, add summary cards for Users/Waitlist/Waitlist Settings
## Context

The operator admin home route (`/admin`) is currently labeled **Admin Dashboard** in the sidebar nav and **Dashboard** on the page itself. It shows a single **Users** card with descriptive copy but no live metrics.

Waitlist and Waitlist Settings already exist as dedicated pages (`/admin/waitlist`, `/admin/waitlist/settings`) in sidebar nav, but they are not represented on the overview. Breadcrumbs also only handle the Users route today — Waitlist and Waitlist Settings fall back to a single **Admin Dashboard** crumb with no section label.

Relevant files today:

- `apps/web/src/admin/pages/AdminDashboardPage.tsx` — overview page (heading + Users card only)
- `apps/web/src/admin/adminNav.ts` — sidebar nav labels (`Admin Dashboard`, `Users`, `Waitlist`, `Waitlist settings`)
- `apps/web/src/admin/AdminLayoutShell.tsx` — `breadcrumbsForPath()` only special-cases `/users`

### 1. Rename "Dashboard" → "Overview"

Update the `/admin` view and nav to use **Overview** instead of **Admin Dashboard** / **Dashboard**:

- Sidebar nav item label: **Overview** (route stays `/admin`)
- Page heading (`AdminDashboardPage`): **Overview**
- Breadcrumb root label: **Overview** (linked to `/admin`)

> Note: The sidebar footer **Dashboard** link (to `/admin/app` user dashboard at `/dashboard`) is separate and should **not** be renamed — that is the escape hatch back to the end-user app (#330).

### 2. Enhance overview cards

Extend the overview card grid (same card pattern as the existing Users card) to show live summary data:

| Card | Route | Summary metric |
|------|-------|----------------|
| **Users** | `/admin/users` | Total user count |
| **Waitlist** | `/admin/waitlist` | Total pending waitlist entries |
| **Waitlist Settings** | `/admin/waitlist/settings` | Current signup mode |

**Users card:** Display total users (e.g. "42 total users"). Data is available via `listUsers()` from `@beakerstack/admin`, which returns `{ total }` on `AdminListUsersResult`.

**Waitlist card:** Display count of **pending** waitlist entries (e.g. "12 pending"). Data is available via `listWaitlistEntries()` from `@beakerstack/waitlist` with `status: 'pending'` — `WaitlistListResult.total`.

**Waitlist Settings card:** Display current signup mode using human-readable labels consistent with the settings page:

- `open` → **Open**
- `waitlist` → **Waitlist**
- `invite_only` → **Invite Only**
- `closed` → **Closed**

Data via `getAdminWaitlistSettings()` from `@beakerstack/waitlist` (`signup_mode` on `WaitlistAdminSettings`).

Cards should handle loading and error states gracefully (skeleton, fallback copy, or muted "—" — match existing admin page patterns).

### 3. Fix breadcrumbs for Waitlist routes

Update `breadcrumbsForPath()` in `AdminLayoutShell.tsx` so breadcrumbs mirror the Users pattern:

- `/admin/users` → `Overview` → `Users`
- `/admin/waitlist` → `Overview` → `Waitlist`
- `/admin/waitlist/settings` → `Overview` → `Waitlist Settings`
- `/admin` → `Overview` (single crumb)

Ensure the **Overview** root crumb links to `/admin`.

## Acceptance criteria

- [ ] Sidebar nav and `/admin` page heading say **Overview** (not "Admin Dashboard" or "Dashboard")
- [ ] Breadcrumb root on all `/admin/*` routes says **Overview** and links to `/admin`
- [ ] Overview **Users** card shows total user count
- [ ] Overview **Waitlist** card links to `/admin/waitlist` and shows total pending waitlist entries
- [ ] Overview **Waitlist Settings** card links to `/admin/waitlist/settings` and shows current signup mode (Open / Waitlist / Invite Only / Closed)
- [ ] Breadcrumbs on `/admin/waitlist` show `Overview → Waitlist`
- [ ] Breadcrumbs on `/admin/waitlist/settings` show `Overview → Waitlist Settings`
- [ ] Breadcrumbs on `/admin/users` show `Overview → Users` (updated from "Admin Dashboard")
- [ ] Sidebar footer **Dashboard** link to `/dashboard` is unchanged
- [ ] Unit tests updated for `AdminDashboardPage`, `AdminLayoutShell`, and `adminNav`

## Implementation notes

- Consider a lightweight overview data hook (e.g. `useAdminOverviewStats`) that fetches counts/settings on mount with `limit: 1` (or minimal params) since only totals are needed
- Extract signup mode display labels to a shared helper if useful (settings page already maps modes in `<option>` labels)
- Card layout: reuse existing grid + `Link` card styling from `AdminDashboardPage.tsx`; add `ClipboardList` and `Settings` icons (already used in `adminNav.ts`)
- File rename (`AdminDashboardPage` → `AdminOverviewPage`) is optional — only do it if it improves clarity without unnecessary churn

## Test plan

1. Sign in as admin and open `/admin` — confirm heading and nav say **Overview**
2. Confirm three summary cards render with live data (Users total, pending waitlist count, signup mode)
3. Click each card — navigates to the correct sub-page
4. Visit `/admin/users`, `/admin/waitlist`, `/admin/waitlist/settings` — confirm breadcrumbs show `Overview → <section>`
5. Click **Overview** breadcrumb — returns to `/admin`
6. Confirm sidebar footer **Dashboard** still links to `/dashboard`
7. Run updated unit tests

### PR DESCRIPTION
**Pull request by sarah-artificerinnovations**
## Summary

- Renames the admin "Dashboard" nav label and page heading to **Overview** (`adminNav.ts`, `AdminDashboardPage.tsx`)
- Adds `useAdminOverviewStats` hook — fetches users total, waitlist pending count, and signup mode in parallel via `Promise.all`
- Extends the overview page with **Waitlist** and **Waitlist Settings** summary cards, each showing a live stat; all three cards show `—` while loading
- Fixes `breadcrumbsForPath` in `AdminLayoutShell.tsx` to cover `/admin/waitlist` and `/admin/waitlist/settings` routes (settings checked before waitlist)
- Updates `adminNav.test.ts`, `AdminDashboardPage.test.tsx`, `AdminLayoutShell.test.tsx`; adds `useAdminOverviewStats.test.tsx`

Closes #338

## Test plan

- [ ] `adminNav` test: labels array now starts with "Overview"
- [ ] `AdminDashboardPage` test: heading "Overview", three card links verified, live stats, loading em-dash state
- [ ] `AdminLayoutShell` test: outlet renders for waitlist and waitlist/settings paths
- [ ] `useAdminOverviewStats` test: fetches all three in parallel, surfaces errors, handles null settings

### CODE REVIEW

File: apps/web/src/admin/hooks/useAdminOverviewStats.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    setError(null);
+
+    Promise.all([
+      listUsers(supabase, { limit: 1, offset: 0, productId: adminProductId }),
+      listWaitlistEntries(supabase, { limit: 1, offset: 0, status: 'pending' }),
+      getAdminWaitlistSettings(supabase),
+    ])
+      .then(([usersResult, waitlistResult, settings]) => {
+        if (cancelled) return;
+        setUsersTotal(usersResult.total);
```
Comment (ZappoMan) [inline apps/web/src/admin/hooks/useAdminOverviewStats.ts]: `listUsers` and `listWaitlistEntries` both return `Promise<... | null>` — they resolve `null` on RPC error/not-found rather than rejecting.

Accessing `.total` on a null result throws inside `.then`, which:
- can leave **partial state** if the first `setState` ran before the throw (e.g. users count set, waitlist still null)
- populates `error` with a `TypeError` instead of degrading gracefully

Suggest null-safe assignment:
```ts
setUsersTotal(usersResult?.total ?? null);
setWaitlistPending(waitlistResult?.total ?? null);
```

And add a hook test mirroring the existing "handles null settings response" case for null list results.

File: apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx [hunk 2]
```diff
@@ -42,11 +42,20 @@ describe('AdminLayoutShell', () => {
     expect(screen.getByText('Users outlet')).toBeInTheDocument();
   });
 
-  it('uses dashboard-only breadcrumbs on non-users admin routes', () => {
+  it('renders outlet for waitlist path', () => {
```
Comment (ZappoMan) [inline apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx:45]: This replaces the old breadcrumb test with an outlet-only assertion. Issue #338 acceptance criteria explicitly require breadcrumb coverage:

- `/admin/users` → `Overview → Users`
- `/admin/waitlist` → `Overview → Waitlist`
- `/admin/waitlist/settings` → `Overview → Waitlist Settings`
- `/admin` → `Overview` (single crumb, no link)

Could extend these tests (or add a dedicated `describe('breadcrumbs')`) using `screen.getByRole('navigation', { name: 'Breadcrumb' })` and asserting link hrefs/text. The outlet checks are fine to keep — just don't drop breadcrumb assertions.

File: apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  });
+
+  it('wraps non-Error throws into Error instances', async () => {
+    mockListWaitlistEntries.mockRejectedValueOnce('boom');
+    const { result } = renderHook(() => useAdminOverviewStats());
+    await waitFor(() => expect(result.current.error).toBeInstanceOf(Error));
+    expect(result.current.error?.message).toBe('boom');
+  });
+
+  it('handles null settings response', async () => {
```
Comment (ZappoMan) [inline apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx:77]: Nice coverage for null settings. Once the hook handles null list RPC responses (see comment on `useAdminOverviewStats.ts`), please add a case here — e.g. `mockListUsers.mockResolvedValue(null)` — asserting `usersTotal` stays `null` and `error` stays `null` (graceful degrade to em-dash on the page).

File: apps/web/src/admin/pages/AdminDashboardPage.tsx [hunk 4]
```diff
@@ -22,8 +37,42 @@ export default function AdminDashboardPage() {
             </span>
             <div>
               <p className='font-semibold text-gray-900'>Users</p>
+              <p className='text-sm text-gray-500'>{stat(usersTotal)} total</p>
```
Comment (ZappoMan) [inline apps/web/src/admin/pages/AdminDashboardPage.tsx]: Nit (non-blocking): the Users card previously had descriptive copy ("View signups, plans, and metered usage") beneath the title. The new cards only show the stat line. That's consistent across all three cards, but if we want a bit more scanability consider keeping a short description under the title and showing the stat on a second line — totally optional.

**Review by ZappoMan — CHANGES_REQUESTED** [review]
## Review summary

Good overall — this matches the approved #338 plan: Overview rename, parallel stats hook, three summary cards, and expanded breadcrumb routing. All four test files pass locally.

**Requesting changes on two items before merge:**

1. **Null-safe RPC results** — `listUsers` and `listWaitlistEntries` resolve to `null` (they don't reject) when the RPC fails or returns not-found. The current `.then` handler can throw when accessing `.total`, which may leave partial state (e.g. users count set but waitlist not) and surfaces a misleading `TypeError` in `error`.

2. **Breadcrumb test coverage** — #338 acceptance criteria call for breadcrumb assertions on Users, Waitlist, and Waitlist Settings routes. The old breadcrumb test was removed and replaced with outlet-only checks; please add breadcrumb nav assertions (e.g. `getByRole('navigation', { name: 'Breadcrumb' })`).

Minor nits left inline — none blocking once the above are addressed.

File: apps/web/src/admin/hooks/useAdminOverviewStats.ts (hunk 1 above)
Comment (sarah-artificerinnovations) [reply apps/web/src/admin/hooks/useAdminOverviewStats.ts]: Fixed. Changed to `usersResult?.total ?? null` and `waitlistResult?.total ?? null` — null resolves degrade cleanly to null state rather than throwing. Added a null-list test case mirroring the null-settings case.

File: apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx (hunk 2 above)
Comment (sarah-artificerinnovations) [reply apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx:45]: Added a `describe("breadcrumbs")` block with four cases covering /users, /waitlist, /waitlist/settings, and the /admin root (single crumb, no link). Uses `screen.getByRole("navigation", { name: "Breadcrumb" })` as you suggested.

File: apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx (hunk 3 above)
Comment (sarah-artificerinnovations) [reply apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx:77]: Done — added `mockListUsers.mockResolvedValue(null as never)` + `mockListWaitlistEntries.mockResolvedValue(null as never)` test asserting both stats stay null with no error.

File: apps/web/src/admin/pages/AdminDashboardPage.tsx (hunk 4 above)
Comment (sarah-artificerinnovations) [reply apps/web/src/admin/pages/AdminDashboardPage.tsx]: Noted — leaving stat-only layout as-is for now; consistent across all three cards and matches the accepted plan scope. Easy follow-up if the team wants descriptions added back.

File: apps/web/src/admin/hooks/useAdminOverviewStats.ts (hunk 1 above)
Comment (ZappoMan) [reply apps/web/src/admin/hooks/useAdminOverviewStats.ts]: ✅ Confirmed fixed in 006775d — optional chaining on both RPC results and the null-list graceful-degrade test both look good.

File: apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx (hunk 2 above)
Comment (ZappoMan) [reply apps/web/src/admin/__tests__/AdminLayoutShell.test.tsx:45]: ✅ Confirmed fixed — the `describe("breadcrumbs")` block covers all four routes from #338, including the root crumb with no link.

File: apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx (hunk 3 above)
Comment (ZappoMan) [reply apps/web/src/admin/__tests__/useAdminOverviewStats.test.tsx:77]: ✅ Confirmed fixed — null list RPC test passes and matches the hook behavior.

File: apps/web/src/admin/pages/AdminDashboardPage.tsx (hunk 4 above)
Comment (ZappoMan) [reply apps/web/src/admin/pages/AdminDashboardPage.tsx]: ✅ Confirmed acceptable — stat-only layout is fine for this PR. The follow-up copy tweak to "total users" is a nice improvement.

File: apps/web/src/admin/pages/AdminDashboardPage.tsx [hunk 5]
```diff
@@ -1,11 +1,26 @@
...
-import { Users } from 'lucide-react';
+import { ClipboardList, Settings, Users } from 'lucide-react';
+import { useAdminOverviewStats } from '../hooks/useAdminOverviewStats';
+import type { SignupMode } from '@beakerstack/waitlist';
+
+const SIGNUP_MODE_LABELS: Record<SignupMode, string> = {
+  open: 'Open',
+  waitlist: 'Waitlist',
+  invite_only: 'Invite Only',
+  closed: 'Closed',
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/web/src/admin/pages/AdminDashboardPage.tsx:10]: ✅ Confirmed fixed — `invite_only` now renders as "Invite only", consistent with the waitlist settings page.

File: apps/web/src/admin/pages/AdminDashboardPage.tsx [hunk 6]
```diff
@@ -1,11 +1,26 @@
...
+};
+
+export default function AdminOverviewPage() {
+  const { usersTotal, waitlistPending, signupMode, loading } =
+    useAdminOverviewStats();
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/web/src/admin/pages/AdminDashboardPage.tsx:15]: Acknowledged — filename/import still say `AdminDashboardPage` while the export is `AdminOverviewPage`. This matches the approved #338 plan (optional rename to avoid churn). Fine to merge as-is; rename can be a follow-up if we want full consistency.

File: apps/web/src/admin/AdminLayoutShell.tsx [hunk 7]
```diff
@@ -6,10 +6,17 @@ import { adminNavItems } from './adminNav';
...
+    return [root, { label: 'Waitlist Settings' }];
   }
-  return [{ label: 'Admin Dashboard' }];
+  if (pathname.includes('/waitlist')) {
+    return [root, { label: 'Waitlist' }];
+  }
+  if (pathname.includes('/users')) {
+    return [root, { label: 'Users' }];
+  }
+  return [{ label: 'Overview' }];
```
[the remark this answers is not in the document]
Comment (ZappoMan) [reply apps/web/src/admin/AdminLayoutShell.tsx:19]: Not a bug for this PR — #338 acceptance criteria specify a single "Overview" crumb with no link on `/admin` (current page). Sub-routes correctly use `{ label: "Overview", to: "/admin" }` as the linked root. Test coverage confirms the intended behavior.

**Review by ZappoMan — APPROVED** [review]
## Re-review — approved

All blocking feedback from the prior review is addressed:

- ✅ Null-safe RPC handling in `useAdminOverviewStats`
- ✅ Breadcrumb test coverage for Users, Waitlist, Waitlist Settings, and Overview root
- ✅ Null list RPC graceful-degrade test
- ✅ Signup mode label casing aligned with settings page ("Invite only")
- ✅ Users stat copy updated to "total users"

Non-blocking items resolved or accepted:
- Stat-only card layout (consistent across cards; matches plan scope)
- Component/file naming mismatch (`AdminOverviewPage` in `AdminDashboardPage.tsx`) — intentional per approved plan

Manually tested build looks good. LGTM.
